tests/unit_tests/wipeable_string.cpp

```diff
@@ -0,0 +1,181 @@
+  ASSERT_EQ(s0.size(), 2);
+  s0.push_back('o');
+  ASSERT_EQ(s0.size(), 3);
+  ASSERT_TRUE(!memcmp(s0.data(), "foo", s0.size()));
+}
+
+TEST(wipeable_string, append_char)
+{
+  epee::wipeable_string s0(std::string("fo"));
+  ASSERT_EQ(s0.size(), 2);
+  s0 += 'o';
+  ASSERT_EQ(s0.size(), 3);
+  ASSERT_TRUE(!memcmp(s0.data(), "foo", s0.size()));
+}
+
+TEST(wipeable_string, append_string)
+{
+  epee::wipeable_string s0(std::string("foo"));
+  ASSERT_EQ(s0.size(), 3);
+  s0 += "bar";
+  ASSERT_EQ(s0.size(), 6);
+  ASSERT_TRUE(!memcmp(s0.data(), "foobar", s0.size()));
+}
+
+TEST(wipeable_string, empty)
+{
+  epee::wipeable_string s0;
+  ASSERT_TRUE(s0.empty());
+  s0.push_back(' ');
+  ASSERT_FALSE(s0.empty());
+  s0.pop_back();
+  ASSERT_TRUE(s0.empty());
+}
+
+TEST(wipeable_string, pop_back)
+{
+  epee::wipeable_string s = "test";
+  ASSERT_EQ(s.size(), 4);
+  s.pop_back();
+  ASSERT_EQ(s.size(), 3);
+  ASSERT_TRUE(!memcmp(s.data(), "tes", s.size()));
+}
+
+TEST(wipeable_string, equal)
+{
+  epee::wipeable_string s0 = "foo";
+  epee::wipeable_string s1 = "bar";
+  epee::wipeable_string s0_2 = "foo";
+  ASSERT_TRUE(s0 == s0);
+  ASSERT_TRUE(s0 == s0_2);
+  ASSERT_TRUE(s1 == s1);
+  ASSERT_FALSE(s1 == s0);
+  ASSERT_FALSE(s1 == s0_2);
+}
+
+TEST(wipeable_string, not_equal)
+{
+  epee::wipeable_string s0 = "foo";
+  epee::wipeable_string s1 = "bar";
+  epee::wipeable_string s0_2 = "foo";
+  ASSERT_FALSE(s0 != s0);
+  ASSERT_FALSE(s0 != s0_2);
+  ASSERT_FALSE(s1 != s1);
+  ASSERT_TRUE(s1 != s0);
+  ASSERT_TRUE(s1 != s0_2);
+}
+
+static epee::wipeable_string trimmed(const char *s)
+{
+  epee::wipeable_string str(s);
+  str.trim();
+  return str;
+}
+
+TEST(wipeable_string, trim)
+{
+  ASSERT_TRUE(trimmed("") == "");
+  ASSERT_TRUE(trimmed(" ") == "");
+  ASSERT_TRUE(trimmed("  ") == "");
+  ASSERT_TRUE(trimmed("a") == "a");
+  ASSERT_TRUE(trimmed(" a") == "a");
+  ASSERT_TRUE(trimmed("  a") == "a");
+  ASSERT_TRUE(trimmed("a ") == "a");
+  ASSERT_TRUE(trimmed("a  ") == "a");
+  ASSERT_TRUE(trimmed(" a  ") == "a");
+  ASSERT_TRUE(trimmed("  a  ") == "a");
+  ASSERT_TRUE(trimmed("  ab  ") == "ab");
+  ASSERT_TRUE(trimmed("  a b  ") == "a b");
+  ASSERT_TRUE(trimmed("  a  b  ") == "a  b");
+}
+
+static bool check_split(const char *s, const std::vector<epee::wipeable_string> &v)
+{
+  epee::wipeable_string str(s);
+  std::vector<epee::wipeable_string> fields;
+  str.split(fields);
+  return v == fields;
+}
+
+TEST(wipeable_string, split)
+{
+  ASSERT_TRUE(check_split("", {}));
+  ASSERT_TRUE(check_split("foo", {"foo"}));
+  ASSERT_TRUE(check_split("foo bar", {"foo", "bar"}));
+  ASSERT_TRUE(check_split("foo     bar", {"foo", "bar"}));
+  ASSERT_TRUE(check_split("foo     bar   baz", {"foo", "bar", "baz"}));
+  ASSERT_TRUE(check_split(" foo     bar   baz         ", {"foo", "bar", "baz"}));
+  ASSERT_TRUE(check_split("  foo     bar   baz", {"foo", "bar", "baz"}));
+  ASSERT_TRUE(check_split("foo     bar   baz ", {"foo", "bar", "baz"}));
+}
```

Comment:
Maybe add a unit test for `parse_hexstr` as well?

Reply:
OK